bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -610,10 +610,29 @@ public Image(Device device, ImageDataProvider imageDataProvider) {
 		SWT.error(SWT.ERROR_INVALID_ARGUMENT, null,
 				": ImageDataProvider [" + imageDataProvider + "] returns null ImageData at 100% zoom.");
 	}
+	if (Device.strictChecks) {
+		validateLinearScaling(imageDataProvider);
+	}
 	init();
 	this.device.registerResourceWithZoomSupport(this);
 }
 
+private void validateLinearScaling(ImageDataProvider provider) {
+	final int baseZoom = 100;
+	final int scaledZoom = 200;
+	final int scaleFactor =  scaledZoom / baseZoom;
+	ImageData baseImageData = provider.getImageData(baseZoom);
+	ImageData scaledImageData = provider.getImageData(scaledZoom);
+
+	if (scaledImageData == null) {
+		return;
+	}
+
+	if (scaledImageData.width != scaleFactor * baseImageData.width || scaledImageData.height != scaleFactor * baseImageData.height) {
+		System.err.println("ImageData should be linearly scaled across zooms.");
```

Comment:
I think we should print the stack trace here, so that someone can more easily find the problematic ImageDataProvider. In addition, we prefix all other strict check prints with `***WARNING: `, so we should probably do the same here.

Reply:
Done